src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx

```diff
@@ -81,8 +81,11 @@ export default class PopoverOverlay extends React.PureComponent<Props, State> {
   componentDidUpdate(oldProps: Props) {
     if (this.props.active && !oldProps.active) {
       this.focusContent();
-
-      this.changeTransitionStatus(TransitionStatus.Entered);
+      this.changeTransitionStatus(TransitionStatus.Entering, () => {
+        setTimeout(() => {
```

Comment:
I think we’d need to ensure this timeout gets cleared if the component gets unmounted?

Reply:
Yeah probably a good practice to do this

Reply:
Totally. I'll clear the newly added `setTimeout` (entering) and the existing one (exiting).